### PR TITLE
Fixes #29450 - Use sp-references in used taxonomy lookup

### PR DESCRIPTION
**Pull request by adamruzicka**
TODOs:
- [ ] ? drop `SmartProxy#taxonomy_foreign_conditions` which is wrong anyway?
- [x] Set up smart proxy references in katello
https://github.com/Katello/katello/pull/10833
- [x] Set up smart proxy references in openscap
https://github.com/theforeman/foreman_openscap/pull/555

### CODE REVIEW

File: app/models/concerns/smart_proxy_host_extensions.rb [hunk 1]
```diff
@@ -7,8 +7,12 @@ def smart_proxy_reference(hash)
       ProxyReferenceRegistry.add_smart_proxy_reference hash
     end
 
+    def smart_proxy_scope(hosts_scope = Host::Managed.all)
```
Comment (ShimShtein) [inline app/models/concerns/smart_proxy_host_extensions.rb]: Nitpick: we can make it a scope:

```suggestion
    scope :with_smart_proxies, -> { eager_load(ProxyReferenceRegistry.smart_proxy_references.select(&:join?).map(&:join_relation) }
```

I have dereferenced the `proxy_join_tables` method. I suppose we can reuse it, if the method will be defined as static.

then later we can use it directly:

``` ruby
hosts_scope.with_smart_proxies.pluck(proxy_column_list).flatten.uniq.compact
```

**Review by ShimShtein — COMMENTED** [review]
If I read correctly, this PR will hide proxies that are not associated with at least one host in the current org/location.
I think it will make non-associated proxies to entirely disappear.

@adamruzicka  Is this your original intention?

**Comment by adamruzicka** [comment]
>  Is this your original intention?

That sounds about right

**Comment by ShimShtein** [comment]
Not entirely sure it's related to this PR, but I see I can give the proxy more visibility than I have:
image

This user has permissions only for `empty_org`, but he can assign also default_org to the proxy, although he is not able to select it in the selector.

**Comment by adamruzicka** [comment]
What exact roles or permissions did you give to that user? If I give the user the org admin role then I can reproduce both with and without this patch. Apparently as long as the user has a role grants them the `assign_organizations` permission, they'll be able to assign organizations which they can't see, which is unfortunate but not really related to what I'm trying to solve here.

**Comment by ShimShtein** [comment]
I have tried it with org_admin role indeed.

**Review by ShimShtein — APPROVED** [review]
@ares This one looks good to me. Any additional concerns before I merge it?

**Review by ofedoren — APPROVED** [review]
Thanks, @adamruzicka and @ShimShtein !

**Comment by ofedoren** [comment]
@adamruzicka, could you please squash the commits?

**Pull request merged**